driver-core/src/main/com/mongodb/client/model/Accumulators.java

```diff
@@ -67,6 +67,44 @@ public static <TExpression> BsonField avg(final String fieldName, final TExpress
         return accumulatorOperator("$avg", fieldName, expression);
     }
 
+    /**
+     * Returns a combination of a computed field and an accumulator that generates a BSON {@link org.bson.BsonType#ARRAY Array}
+     * containing computed values from the given {@code inExpression} based on the provided {@code pExpression}, which represents an array
+     * of percentiles of interest within a group, where each element is a numeric value between 0.0 and 1.0 (inclusive).
+     *
+     * @param fieldName      The field computed by the accumulator.
+     * @param inExpression   The input expression.
+     * @param pExpression    The expression representing a percentiles of interest.
+     * @param method         The method to be used for computing the percentiles.
+     * @param <InExpression> The type of the input expression.
+     * @param <PExpression>  The type of the percentile expression.
+     * @return The requested {@link BsonField}.
+     * @mongodb.driver.manual reference/operator/aggregation/percentile/ $percentile
+     * @since 4.10
+     * @mongodb.server.release 7.0
+     */
+    public static <InExpression, PExpression> BsonField percentile(final String fieldName, final InExpression inExpression,
```

Comment:
Currently, not all array expressions are supported in MQL.

Additionally, if we start incorporating MQLValue explicitly into the Accumulators builder class for new methods, it would introduce a combination of the old and new approaches. Some methods would accept Object/Bson, while others would accept MQL. This could potentially confuse users, especially those who are still using the Document/BsonDocument approach in their code.

In my opinion, it would be better to wait until aggregations are available in the computational API, where we could use only MQL entities.